docs/configuration/config-file-reference.md

```diff
@@ -3724,30 +3724,31 @@ The `memberlist_config` configures the Gossip memberlist.
 [randomize_node_name: <boolean> | default = true]
 
 # The timeout for establishing a connection with a remote node, and for
-# read/write operations. Uses memberlist LAN defaults if 0.
+# read/write operations. Uses default if set to 0.
```

Comment:
This sounds weird.  Do we still want it?

Reply:
On reflection it is probably better than an error, or actually doing a timeout immediately on starting the operation.
Still, I was expecting this PR to remove the lines of code doing `if x == 0 ...`

Reply:
> To prevent this being a breaking change, the existing behaviour is
retained, in case anyone is explicitly setting the values to zero and
expecting the default to be used.

Happy to remove them if there's a consensus to do so.

Reply:
I am in favor of removing support for 0 values.

Reply:
I'm also in favour of removing support for 0 values (assuming will be documented in the CHANGELOG as CHANGE).

Reply:
Done - was a bit more complicated as the tests relied on the default values.